script.js

```diff
@@ -1,4 +1,183 @@
+const resetBtn = document.querySelector("#reset-button");
+resetBtn.style.display = "none";
+
+let gameRecordArr = [];
+let playerCounter = 1;
+let diceRollMsg, winner, winnerMsg;
+let displayMsg = "";
+let numsArr = "";
+let gameBoard;
+
+// Toggle Buttons
+function toggleBtn() {
+  if (resetBtn.style.display == "none") {
+    submitBtn.style.display = "none";
+    resetBtn.style.display = "block";
+  } else {
+    submitBtn.style.display = "block";
+    resetBtn.style.display = "none";
+  }
+}
+
+// Submit button click event
+submitBtn.addEventListener("click", function () {
+  let result = main();
+  submitBtn.disable = true;
+  outputDiv.innerHTML = result;
+});
+
+// Reset button click event
+resetBtn.addEventListener("click", function () {
+  location.reload();
+});
+
+// Roll dice
+function rollDice() {
+  let diceNum = Math.floor(Math.random() * 6) + 1;
+  return diceNum;
+}
+
+// Bubble Sort: Array low --> high or high --> low
+function sortDiceNumArr(arr, mode) {
+  const swap = (arr, index1, index2) => {
+    [arr[index1], arr[index2]] = [arr[index2], arr[index1]];
+  };
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr.length - i; j++) {
+      if (arr[j] < arr[j + 1] && mode == "big") {
+        swap(arr, j, j + 1);
+      } else if (arr[j] > arr[j + 1] && mode == "small") {
+        swap(arr, j, j + 1);
+      }
+    }
+  }
+  return arr;
+}
+
+// Get the biggest/lowest dice number combination
+function getDiceCombo(num_of_dice, mode) {
+  let counter = 0;
+  let arr = [];
+  while (counter < Number(num_of_dice)) {
+    arr.push(rollDice());
+    counter += 1;
+  }
+  numsArr = arr.toString();
+  let sortedArr = sortDiceNumArr(arr, mode);
+  let numStr = "";
+  for (let i = 0; i < sortedArr.length; i++) {
+    numStr += String(sortedArr[i]);
+  }
+  let combination = Number(numStr);
+  return combination;
+}
+
+// Generate a sorted leaderboard to show each player's optimal number
+function generateLeaderboard(arr, mode) {
+  let record;
+  let showRecord = "";
+  let leaderboard = [];
+  for (let i = 0; i < arr.length; i++) {
+    let player = `Player ${i + 1}`;
+    leaderboard.push({ player });
+    leaderboard[i].value = arr[i];
+  }
+  if (mode == "big") {
+    record = leaderboard.sort((a, b) => (a.value < b.value ? 1 : -1));
+  } else {
+    record = leaderboard.sort((a, b) => (a.value > b.value ? 1 : -1));
+  }
+  for (let j = 0; j < record.length; j++) {
+    showRecord +=
+      `${[j + 1]}. ` + record[j].player + " : " + record[j].value + `<br>`;
+  }
+  return showRecord;
+}
+
+// Find the index of duplicated number in game board array
+function findDuplicatesIndex(arr, num) {
+  let duplicatesIndex = [];
+  let idx = arr.indexOf(num);
+  while (idx != -1) {
+    duplicatesIndex.push(idx);
+    idx = arr.indexOf(num, idx + 1);
+  }
+  return duplicatesIndex;
+}
+
+// Check winner and display
+function checkWinner(numOfDice, numOfPlayer, gameMode) {
```

Comment:
this function is a bit lengthy and can prob be refactored, seems like its doing more than just check the winner

your whole gameplay logic is housed inside here too, so this can prob be broken up further

Reply:
noticed you wanted more specific feedback in this area:

this function is currently doing many things - 
1. roll dice
2. generate leaderboard (this is already abstracted to a helper function)
3. check for dulNum (maybe we can merge this with findDuplicatesIndex?)
4. output winner msg

this will be a nice starting point on refactoring this further into smaller functions